core/common/src/main/java/alluxio/security/authentication/SaslStreamClientDriver.java

```diff
@@ -41,20 +42,24 @@
   private SaslHandshakeClientHandler mSaslHandshakeClientHandler;
   /** Used to wait until authentication is completed. */
   private SettableFuture<Boolean> mAuthenticated;
+  /** Whether the authentication is active. */
```

Comment:
can you add comments on how this is different from `mAuthenticated` and how they interact? Are both required? Having both is very confusing.

Reply:
SettableFuture is unfortunately not resettable, so it can't be used to reflect state changes following the initial state. I changed names and comments to try to make it less confusing.